app/bundles/EmailBundle/Controller/Api/EmailApiController.php

```diff
@@ -163,4 +177,136 @@ public function sendLeadAction($id, $leadId)
 
         return $this->notFound();
     }
+
+    /**
+     * Sends custom content to a specific lead.
+     *
+     * @param int $contactId
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
+     */
+    public function sendCustomLeadAction($contactId)
```

Comment:
There is a lot of similar / duplicate code in old and new methods. Controller should not hold the logic. Could you move it to a new class and try to refactor it, so we do not have so many duplicate code in separate methods?
Would be great to clean an old methods too, because they share some logic.

Reply:
Yes, I can. 
Can I move it to helper? 
Or do you have any suggest?

Reply:
It would be great if you can create a service and move logic there.
I think there can be another service for getting parameters from request. This approach should eliminate duplicate code.